src/widget_window.c

```diff
@@ -113,11 +113,60 @@ GtkWidget *widget_window_create(
 		else
 			g_warning("%s(): Unknown layer %s.", __func__, value);
 	}
-
-	if (layer != GTK_LAYER_SHELL_LAYER_ENTRY_NUMBER) {
+	
+	GtkLayerShellEdge edge = GTK_LAYER_SHELL_EDGE_ENTRY_NUMBER;
+	GtkLayerShellEdge corner = GTK_LAYER_SHELL_EDGE_ENTRY_NUMBER;
+	
+	value = get_tag_attribute(attr, "edge");
+	if (value) {
+		if (strcmp(value, "top") == 0)
+			edge = GTK_LAYER_SHELL_EDGE_TOP;
+		else if (strcmp(value, "bottom") == 0)
+			edge = GTK_LAYER_SHELL_EDGE_BOTTOM;
+		else if (strcmp(value, "left") == 0)
+			edge = GTK_LAYER_SHELL_EDGE_LEFT;
+		else if (strcmp(value, "right") == 0)
+			edge = GTK_LAYER_SHELL_EDGE_RIGHT;
+		else if (strcmp(value, "topleft") == 0) {
+			edge = GTK_LAYER_SHELL_EDGE_LEFT;
+			corner = GTK_LAYER_SHELL_EDGE_TOP;
+		}
+		else if (strcmp(value, "bottomleft") == 0) {
+			edge = GTK_LAYER_SHELL_EDGE_LEFT;
+			corner = GTK_LAYER_SHELL_EDGE_BOTTOM;
+		}
+		else if (strcmp(value, "topright") == 0) {
+			edge = GTK_LAYER_SHELL_EDGE_RIGHT;
+			corner = GTK_LAYER_SHELL_EDGE_TOP;
+		}
+		else if (strcmp(value, "bottomright") == 0) {
+			edge = GTK_LAYER_SHELL_EDGE_RIGHT;
+			corner = GTK_LAYER_SHELL_EDGE_BOTTOM;
+		}
+		else
+			g_warning("%s(): Unknown edge %s.", __func__, value); 
+	}
+	
+	if ((layer != GTK_LAYER_SHELL_LAYER_ENTRY_NUMBER) ||
+		(edge != GTK_LAYER_SHELL_EDGE_ENTRY_NUMBER)  ||
+		(corner != GTK_LAYER_SHELL_EDGE_ENTRY_NUMBER)) {
 		gtk_layer_init_for_window(GTK_WINDOW(widget));
+	}
+	if (layer != GTK_LAYER_SHELL_LAYER_ENTRY_NUMBER) {
 		gtk_layer_set_layer(GTK_WINDOW(widget), layer);
 	}
+	if (edge != GTK_LAYER_SHELL_EDGE_ENTRY_NUMBER) {
+		gtk_layer_auto_exclusive_zone_enable (GTK_WINDOW(widget));
+		gtk_layer_set_margin(GTK_WINDOW(widget), GTK_LAYER_SHELL_EDGE_LEFT, 20);
```

Comment:
20 multiplied by the monitor's scaling factor or just 20?

Reply:
I guess we'll have to ask William Wold, I don't see any ref in the sources.

Reply:
20 makes more sense, because we don't know what monitor the window is going to be displayed on, and it can span multiple monitors. IMHO we should merge, and fix scaling issues later (if needed).